File: WalletTracker/app/src/main/java/com/chtima/wallettracker/components/ShadowConstraintLayout.java
package com.chtima.wallettracker.components;

import android.content.Context;
import android.util.AttributeSet;
import android.util.TypedValue;
import android.view.ViewTreeObserver;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;

import com.chtima.wallettracker.R;

public class ShadowConstraintLayout extends ConstraintLayout {
    public ShadowConstraintLayout(@NonNull Context context) {
        super(context);
        init(context);
    }

    public ShadowConstraintLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public ShadowConstraintLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    public ShadowConstraintLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init(context);
    }

    private void init(Context context){
        int i = 0;

        getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
            @Override
            public void onGlobalLayout() {
                if (getViewTreeObserver().isAlive())
                    getViewTreeObserver().removeOnGlobalLayoutListener(this);

                ConstraintLayout.LayoutParams params = (ConstraintLayout.LayoutParams) getLayoutParams();
                params.setMargins(dpToPx(16), dpToPx(16), dpToPx(16), dpToPx(16));
                setBackgroundResource(R.drawable.round_layout);
                setElevation(dpToPx(5));
                setLayoutParams(params);
            }
        });

    }

    private int dpToPx(float dp) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, getResources().getDisplayMetrics());
    }

}

File: WalletTracker/app/src/main/java/com/chtima/wallettracker/components/ShadowConstraintLayout.kt
package com.chtima.wallettracker.components

import android.content.Context
import android.util.AttributeSet
import android.util.TypedValue
import android.view.ViewOutlineProvider
import android.view.ViewTreeObserver
import androidx.annotation.AttrRes
import androidx.constraintlayout.widget.ConstraintLayout
import com.chtima.wallettracker.R

class ShadowConstraintLayout @JvmOverloads constructor(
    context: Context,
    attrRes: AttributeSet? = null,
    defStyleAttr: Int = 0,
    defStyleRes: Int = R.style.ShadowConstraintLayoutStyle
) : ConstraintLayout(context, attrRes, defStyleAttr, defStyleRes) {

    init {
        getViewTreeObserver().addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener{
            override fun onGlobalLayout() {
                if (viewTreeObserver.isAlive)
                    viewTreeObserver.removeOnGlobalLayoutListener(this)

                val params = layoutParams as ConstraintLayout.LayoutParams;
                params.setMargins(dpToPx(16f), dpToPx(16f), dpToPx(16f), dpToPx(16f))
                elevation = dpToPx(10f).toFloat()
                outlineProvider = ViewOutlineProvider.BACKGROUND
                clipToOutline = true
                setLayoutParams(params)
            }
        })
    }

    fun dpToPx(dp: Float):Int{
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, resources.displayMetrics).toInt()
    }


}